src/common/otp.c

```diff
@@ -13,3 +13,7 @@ const char *otp_get_mac_address_str() {
     }
     return mac_address_str;
 }
+
+const char *otp_get_serial_number() {
+    return (const char *)(OTP_SERIAL_NUMBER_ADDR);
```

Comment:
I'm not sure this is valid. Isn't this a cast from volatile to non-volatile?

Reply:
fixed